Merge Rewrite into Master

Asherslab opened the pull request:
This PR is where ChatChain begins, once this is ready and merged it will be version 1.0 of the ChatChain service!

Features so far:
1. A WebApp for managing clients and groups.
2. An IdentityServer4 project for authenticating those clients.
3. The SignalR hub setup for:
    1. Generic text messages with a user, group, client, and etc.
    2. Retrieving a client's available groups from the server. 
    3. Retrieving a client's server-side client.
    4. Client event messages.
4. MongoDB data storage for everything.
5. Redis backplane setup for load balancing.
6. Environment Variable configuration for all needed values

Asherslab committed:
init

Asherslab committed:
Revert "Merge remote-tracking branch 'origin/master' into event-changes"

This reverts commit ee869e547d0bd505a2d7b84f32d4b82de81ff418.

Asherslab committed:
Merge remote-tracking branch 'origin/master' into event-changes

Asherslab committed:
Server <-> Server communication

Asherslab committed:
changes

Asherslab committed:
add simple startup script

Asherslab committed:
global.json

Asherslab committed:
Update Dockerfile

Asherslab committed:
Update Dockerfile

Asherslab committed:
Update appsettings.json

Asherslab committed:
Update appsettings.json

Asherslab committed:
change hub name

Asherslab committed:
update to 2.1.301

Asherslab committed:
Update Dockerfile

Asherslab committed:
rename

Asherslab committed:
update sdk

Asherslab committed:
testing

Asherslab committed:
init

Asherslab committed:
commit

Asherslab committed:
rewriting and stuff

Asherslab committed:
commit

Asherslab committed:
commit

Asherslab committed:
commit

Asherslab committed:
Debug

Asherslab committed:
testing

Asherslab committed:
Push

Asherslab committed:
Move to MongoDB

Asherslab committed:
Moved to MongoDB and Fully Functioning!

Asherslab committed:
Docker added

Asherslab committed:
whoops.....

Asherslab committed:
Ocelot and more docker

Asherslab committed:
changes

Asherslab committed:
docker testing

Asherslab committed:
commit

Asherslab committed:
commit

Asherslab committed:
coomit

Asherslab committed:
remove some html stuff

Asherslab committed:
redis backplane

Asherslab committed:
wait

Asherslab committed:
docker wait

Asherslab committed:
test

Asherslab committed:
test

Asherslab committed:
test2

Asherslab committed:
test3

Asherslab committed:
PublicOrigin

Asherslab committed:
testing

Asherslab committed:
testing2

Asherslab committed:
testing3

Asherslab committed:
testing

Asherslab committed:
testing2

Asherslab committed:
testing3

Asherslab committed:
testing4

Asherslab committed:
testing5

Asherslab committed:
testing6

Asherslab committed:
testing7

Asherslab committed:
testing8

Asherslab committed:
identityserver mongodb testing

Asherslab committed:
testing1

Asherslab committed:
testing2

Asherslab committed:
testing3

Asherslab committed:
testing4

Asherslab committed:
testing5

Asherslab committed:
testing6

Asherslab committed:
testing7 wtf

Asherslab committed:
testing8

Asherslab committed:
testing9 holy cow

Asherslab committed:
testing10

Asherslab committed:
testing11

Asherslab committed:
testing12

Asherslab committed:
testing13

Asherslab committed:
testing14

Asherslab committed:
testing15

Asherslab committed:
testing16

Asherslab committed:
testing17 -- possible revert

Asherslab committed:
fixy fix

Asherslab committed:
testing18

Asherslab committed:
testing19 -- possible revert

Asherslab committed:
testing20

Asherslab committed:
testing21

Asherslab committed:
testing22

Asherslab committed:
testing23

Asherslab committed:
testing24 -- dumbo

Asherslab committed:
testing15

Asherslab committed:
testing16

Asherslab committed:
testing25

Asherslab committed:
testing26

Asherslab committed:
testing27

Asherslab committed:
testing28

Asherslab committed:
testing29

Asherslab committed:
testing30

Asherslab committed:
testing31

Asherslab committed:
testing32

Asherslab committed:
testing33

Asherslab committed:
testing34

Asherslab committed:
testing35

Asherslab committed:
testing36

Asherslab committed:
testing37 -- previous works btw

Asherslab committed:
testing38

Asherslab committed:
testing39

Asherslab committed:
testing40

Asherslab committed:
testing41

Asherslab committed:
testing42

Asherslab committed:
testing43

Asherslab committed:
testing44

Asherslab committed:
testing45

Asherslab committed:
test46

Asherslab committed:
testing47

Asherslab committed:
testing48

Asherslab committed:
testing49

Asherslab committed:
testing

Asherslab committed:
testing50

Asherslab committed:
testing51

Asherslab committed:
testing52

Asherslab committed:
testing53

Asherslab committed:
testing54

Asherslab committed:
5

Asherslab committed:
testing56

Asherslab committed:
testing57

Asherslab committed:
testing58

Asherslab committed:
testing59

Asherslab committed:
testing60

Asherslab committed:
testing61

Asherslab committed:
testing62

Asherslab committed:
testing63

Asherslab committed:
Cleanup the code a bunch

Asherslab committed:
Change methods and names to be more meaningful

Asherslab committed:
Allow for Extra data to be given with Client and User events

Asherslab committed:
Add Client and Group Description fields - Testing 1

Asherslab committed:
Hotfix client/groups

Asherslab committed:
Client/Groups files extra comment

Asherslab committed:
Allow server to work without an email sender setup

Asherslab committed:
Fix last commit maybe

Asherslab committed:
Again try and fix last commit

Asherslab committed:
Rename IdentityServer WebApp to just WebApp

Asherslab committed:
Fix for previous

Asherslab committed:
Delete .DS_Store

Asherslab committed:
Delete global.json

Asherslab committed:
Update ChatChainHub.cs

Asherslab committed:
Update ChatChainHub.cs

Asherslab committed:
Update ChatChainHub.cs

Asherslab committed:
Update ChatChainHub.cs

Asherslab committed:
Update ChatChainHub.cs

Asherslab committed:
Update ChatChainHub.cs

Asherslab committed:
Update ChatChainHub.cs

Asherslab committed:
Update Index.cshtml

Asherslab committed:
Delete .DS_Store

Asherslab committed:
Merge pull request #3 from ldtteam/Test-Branch

Delete .DS_Store

Asherslab committed:
Delete .DS_Store

Asherslab committed:
Merge pull request #4 from ldtteam/test-pr-4

Delete .DS_Store

Asherslab committed:
Fix Namespaces

Asherslab committed:
Test One

Asherslab committed:
Fix 1

Asherslab committed:
Test two

Asherslab committed:
fix

Asherslab committed:
fix 2

Asherslab committed:
fix 3

Asherslab committed:
test Three

Asherslab committed:
test Four

Asherslab committed:
test Four - fix

Asherslab committed:
test Four - fix

Asherslab committed:
test Four - fix - that was dumb

Asherslab committed:
test Five

Asherslab committed:
test Six

Asherslab committed:
test Seven

Asherslab committed:
test Seven - fix

Asherslab committed:
test Eight

Asherslab committed:
test nine -- make it look prettier

Asherslab committed:
Test 10 with Hub

Asherslab committed:
Update Startup.cs

Asherslab committed:
Update Startup.cs

Asherslab committed:
Update Startup.cs

Asherslab committed:
Update Startup.cs

Asherslab committed:
Update Startup.cs

Asherslab committed:
Update README.md

Asherslab committed:
Update ChatChainHub.cs

Asherslab committed:
better wording

Asherslab committed:
Update README.md

Asherslab committed:
Update README.md

Asherslab committed:
remove Unused object

Asherslab committed:
Merge remote-tracking branch 'origin/features/clientconfigs' into features/clientconfigs

Asherslab committed:
remove Unused object and a couple comments

Asherslab committed:
Merge branch 'rewrite' into features/clientconfigs

Asherslab committed:
Merge pull request #5 from ldtteam/features/clientconfigs

New Client Configs Feature

Asherslab committed:
Client/User Ranks feature (#7)

* Add ClientRank to User
* Add Nickname and Colour objects to User

LDTTeam-Buildsystem commented:
This PR has been automatically built and hosted by our build system, you can access it here: https://2.chatchain.co

AnotherFoxGuy committed:
Website redesign (#29)

Redesign WebApp for new style and layout.
Major Thanks to:
* AnotherFoxGuy
* PoroUsedSnax
* SnaddyVitch-Dispenser
* tobias-g

LDTTeam-Buildsystem commented:
This PR has been automatically built and hosted by our build system, you can access it here: https://2.chatchain.co

Asherslab committed:
Seperate Accounts from WebApp (#31)

* Seperate .Net Core Identity out of WebApp and into IdentityServer4.
* Create ChatChainCommon lib to store files and such between services.
* Switch configuration to IOptions instead of plain environment variables
* Fix stuff and stuff

LDTTeam-Buildsystem commented:
This PR has been automatically built and hosted by our build system, you can access it here: https://2.chatchain.co

Asherslab committed:
Organisations! (#32)

Moves clients and groups into organisations instead of users, allows inviting other users into an organisation, permissions, etc.

LDTTeam-Buildsystem commented:
This PR has been automatically built and hosted by our build system, you can access it here: https://2.chatchain.co

Asherslab committed:
Hotfix Permissions display and saving

LDTTeam-Buildsystem commented:
This PR has been automatically built and hosted by our build system, you can access it here: https://2.chatchain.co

Asherslab committed:
[Draft] Clean Architecture and Extract out API (#33)

LDTTeam-Buildsystem commented:
This PR has been automatically built and hosted by our build system, you can access it here: https://2.chatchain.co

Asherslab committed:
Hotfix NPEs hopefully (#35)

* Update UserEventUseCase.cs

* Update ClientEventUseCase.cs

Asherslab committed:
Update Dockerfile (#36)

* Update Dockerfile

* Update Dockerfile

* Update Dockerfile

* Update Dockerfile

Asherslab committed:
Feature: Stats (#34)

* Working on Auth and Docker-Services-

* Adding services to the docker-compose file

* fix Dockerfile

* Update README.md

* update to 2.1.301

* Updating to newer ASP.Net core packages and Container references

* Fixing discordian server compile via docker.

* Working!

* Clean Architecture working!

* Rework Hub and hopefully get everything working

* All dockerfiles setup and fixed

* Should have fixed everything

* Codacy 1

* Codacy 2

* Init, should be working

* Update Dockerfile

* fix runtime error

* allow for requests by group as well

* fix stats request

* Send Client object along with StatsResponse

* init

* Revert "Merge remote-tracking branch 'origin/master' into event-changes"

This reverts commit ee869e547d0bd505a2d7b84f32d4b82de81ff418.

* Server <-> Server communication

* changes

* add simple startup script

* global.json

* Update Dockerfile

* Update Dockerfile

* Update appsettings.json

* Update appsettings.json

* change hub name

* update to 2.1.301

* Update Dockerfile

* rename

* update sdk

* testing

* init

* commit

* rewriting and stuff

* commit

* commit

* commit

* Debug

* testing

* Push

* Move to MongoDB

* Moved to MongoDB and Fully Functioning!

* Docker added

* whoops.....

* Ocelot and more docker

* changes

* docker testing

* commit

* commit

* coomit

* remove some html stuff

* redis backplane

* wait

* docker wait

* test

* test

* test2

* test3

* PublicOrigin

* testing

* testing2

* testing3

* testing

* testing2

* testing3

* testing4

* testing5

* testing6

* testing7

* testing8

* identityserver mongodb testing

* testing1

* testing2

* testing3

* testing4

* testing5

* testing6

* testing7 wtf

* testing8

* testing9 holy cow

* testing10

* testing11

* testing12

* testing13

* testing14

* testing15

* testing16

* testing17 -- possible revert

* fixy fix

* testing18

* testing19 -- possible revert

* testing20

* testing21

* testing22

* testing23

* testing24 -- dumbo

* testing15

* testing16

* testing25

* testing26

* testing27

* testing28

* testing29

* testing30

* testing31

* testing32

* testing33

* testing34

* testing35

* testing36

* testing37 -- previous works btw

* testing38

* testing39

* testing40

* testing41

* testing42

* testing43

* testing44

* testing45

* test46

* testing47

* testing48

* testing49

* testing

* testing50

* testing51

* testing52

* testing53

* testing54

* 5

* testing56

* testing57

* testing58

* testing59

* testing60

* testing61

* testing62

* testing63

* Cleanup the code a bunch

* Change methods and names to be more meaningful

* Allow for Extra data to be given with Client and User events

* Add Client and Group Description fields - Testing 1

* Hotfix client/groups

* Client/Groups files extra comment

* Allow server to work without an email sender setup

* Fix last commit maybe

* Again try and fix last commit

* Rename IdentityServer WebApp to just WebApp

* Fix for previous

* Delete .DS_Store

* Delete global.json

* Update ChatChainHub.cs

* Update ChatChainHub.cs

* Update ChatChainHub.cs

* Update ChatChainHub.cs

* Update ChatChainHub.cs

* Update ChatChainHub.cs

* Update ChatChainHub.cs

* Update Index.cshtml

* Delete .DS_Store

* Delete .DS_Store

* Fix Namespaces

* Test One

* Fix 1

* Test two

* fix

* fix 2

* fix 3

* test Three

* test Four

* test Four - fix

* test Four - fix

* test Four - fix - that was dumb

* test Five

* test Six

* test Seven

* test Seven - fix

* test Eight

* test nine -- make it look prettier

* Test 10 with Hub

* Update Startup.cs

* Update Startup.cs

* Update Startup.cs

* Update Startup.cs

* Update Startup.cs

* Update README.md

* Update ChatChainHub.cs

* better wording

* remove Unused object

* remove Unused object and a couple comments

* Update README.md

* Update README.md

* Client/User Ranks feature (#7)

* Add ClientRank to User
* Add Nickname and Colour objects to User

* Website redesign (#29)

Redesign WebApp for new style and layout.
Major Thanks to:
* AnotherFoxGuy
* PoroUsedSnax
* SnaddyVitch-Dispenser
* tobias-g

* Seperate Accounts from WebApp (#31)

* Seperate .Net Core Identity out of WebApp and into IdentityServer4.
* Create ChatChainCommon lib to store files and such between services.
* Switch configuration to IOptions instead of plain environment variables
* Fix stuff and stuff

* Organisations! (#32)

Moves clients and groups into organisations instead of users, allows inviting other users into an organisation, permissions, etc.

* Hotfix Permissions display and saving

* Working!

* Clean Architecture working!

* Rework Hub and hopefully get everything working

* All dockerfiles setup and fixed

* Should have fixed everything

* Codacy 1

* Codacy 2

* Init, should be working

* Update Dockerfile

* fix runtime error

* allow for requests by group as well

* fix stats request

* Send Client object along with StatsResponse

* commit

* working

* fixes

* fix docker files

Co-authored-by: Marc Hermans <OrionDevelopment@outlook.com>
Co-authored-by: Edgar <Edgar@AnotherFoxGuy.com>

Asherslab committed:
Update UserEventUseCase.cs (#37)

* Update UserEventUseCase.cs

* Update ClientEventUseCase.cs
